php_array.h

```diff
@@ -355,20 +355,22 @@ char *php_array_zval_to_string(zval *z, int *plen, zend_bool *pfree) {
 	*plen = 0;
 	*pfree = 0;
 	if (!z) { return NULL; }
-	*pfree = 1;
 	switch (Z_TYPE_P(z)) {
 		case IS_NULL:
-			*pfree = 0;
-			return "";
+			return (char *)"";
 		case IS_STRING:
-			*pfree = 0;
 			*plen = Z_STRLEN_P(z);
 			return Z_STRVAL_P(z);
 		default:
 		{
 			zval c = *z;
 			zval_copy_ctor(&c);
 			convert_to_string(&c);
+#ifdef ZEND_ENGINE_3
+			*pfree = ! IS_INTERNED(Z_STR(c));
+#else
+			*pfree = ! IS_INTERNED(Z_STRVAL(c));
```

Comment:
Indeed, this was [introduced in 5.4](https://php-lxr.adamharvey.name/source/xref/PHP-5.4/Zend/zend_string.h#37). I wasn't aware that you were otherwise compatible with 5.3, as we only build with 5.4+.

I don't think we care, but it'd be trivial to fix this later with an extra `#elif` if someone complains :)